src/Codeception/Test/Cept.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Codeception\Test;
+
+use Codeception\Exception\TestParseException;
+use Codeception\Lib\Parser;
+
+class Cept extends Test implements Interfaces\Plain, Interfaces\ScenarioDriven, Interfaces\Reported
+{
+    use Feature\ScenarioLoader;
+
+    /**
+     * @var Parser
+     */
+    protected $parser;
+
+    public function __construct($name, $file)
+    {
+        $this->setMetadata(new Metadata());
+        $this->getMetadata()->setName($name);
+        $this->getMetadata()->setFilename($file);
+        $this->createScenario();
+        $this->parser = new Parser($this->getScenario(), $this->getMetadata());
```

Comment:
How about assigning metadata to local variable and passing it to all methods?
$metadata = new Metadata();
$metadata->setName($name);
$metadata->setFilename($file);
$this->setMetadata($metadata);